tardis/montecarlo/src/integrator.c

```diff
@@ -167,19 +168,26 @@ _formal_integral(
   // global read-only values
   int64_t size_line = storage->no_of_lines,
           size_shell = storage->no_of_shells,
-          size_tau = size_line * size_shell;
+          size_tau = size_line * size_shell,
+          finished_nus = 0;
 
   double R_ph = storage->r_inner[0];
   double R_max = storage->r_outer[size_shell - 1];
   double pp[N];
   double *exp_tau = calloc(size_tau, sizeof(double));
-#pragma omp parallel firstprivate(L, exp_tau)
+#ifdef WITHOPENMP
```

Comment:
For the formal integral openmp parallelization I chose a different approach. By providing a dummy header, we can limit `#ifdef WITHOPENMP` to the header part of the document and I would strongly advise against including it anywhere in the body.

I know this implementation is basically a copy/paste of my progress from `cmontecarlo.c` but I think we are now able to provide a more readable solution.